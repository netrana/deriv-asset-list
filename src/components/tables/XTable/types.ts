import { AlignType } from 'rc-table/lib/interface';
import { FC, ReactNode } from 'react';

export type CellType =
  | 'text'
  | 'image-and-text-cell'
  | 'last-price-cell'
  | 'daily-change-cell'
  | 'mini-chart-cell';
export interface Data {
  key?: string;
}
export interface RenderedCell {
  props?: {
    className?: string;
    style?: React.CSSProperties;
    children?: React.ReactNode;
    colSpan?: number;
    rowSpan?: number;
  };
  children?: React.ReactNode;
}

export interface Column {
  name: string;
  type: CellType;
  key?: string;
  dataIndex?: string;
  render?: (
    value: any, // eslint-disable-line
    record: any, // eslint-disable-line
    index: number
  ) => ReactNode | RenderedCell;
  width?: string | number;
  tooltip?: string;
  align?: AlignType;
}

export interface Cells {
  [index: string]: FC<any>; // eslint-disable-line
}

export interface Props<T extends Data, K> {
  data: T[];
  columns: Column[];
  empty?: ((componentName?: string) => ReactNode) | ReactNode;
  tableClassName?: string;
  rowClassName?: string;
  loading?: boolean;
  showPointerOnRowHover?: boolean;
}
