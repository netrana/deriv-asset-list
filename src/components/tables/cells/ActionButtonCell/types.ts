export interface LabeledValue {
  label: React.ReactNode;
  onClick: () => void;
  btnType?: string;
  icon?: JSX.Element;
}

export interface Props {
  useHook: Function;
  buttons: LabeledValue[];
}
