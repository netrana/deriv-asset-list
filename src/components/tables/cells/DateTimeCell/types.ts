export interface Props {
  date: string;
}
