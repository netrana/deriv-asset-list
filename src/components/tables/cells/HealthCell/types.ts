import { ProgressProps } from 'antd/lib/progress';
import { TagProps } from 'antd/lib/tag';

export interface HealthTagProps {
  name?: string;
  color?: TagProps['color'];
  linkTo?: string;
}
export interface HealthGroupProps {
  name: string;
  tags: HealthTagProps[];
}
export interface HealthTagsProps {
  tags?: HealthTagProps[];
  groups?: HealthGroupProps[];
}

export interface Props extends HealthTagsProps {
  progressPercent?: number;
  progressStatus?: ProgressProps['status'];
}
