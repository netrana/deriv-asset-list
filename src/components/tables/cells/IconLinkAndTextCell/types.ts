import { ReactNode } from 'react';

export interface Row {
  text?: string;
  icon?: ReactNode;
  iconTo: string;
  textTo: string;
  isLinkable?: boolean;
  color?: string;
}
export interface Props {
  rows: Row[];
}
