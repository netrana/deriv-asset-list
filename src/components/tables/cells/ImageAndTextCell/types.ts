import { ReactNode } from "react";

export interface Props {
  text?: string;
  image?: ReactNode;
}
// export interface Props {
//   rows: Row[];
// }
