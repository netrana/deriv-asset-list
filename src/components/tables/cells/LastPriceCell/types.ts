export interface Props {
  symbol: string;
}
