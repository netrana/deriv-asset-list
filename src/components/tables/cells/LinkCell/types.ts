export interface Props {
  text: string;
  link?: string;
}
