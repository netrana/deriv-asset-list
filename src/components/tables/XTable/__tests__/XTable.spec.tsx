import React from 'react';
import render from 'react-test-renderer';

import NXTable from '../XTable';
import { Column } from '../types';

jest.mock('components/helpers', () => {
  const actual = jest.requireActual('components/helpers');
  return {
    ...actual,
    Table: 'Table',
  };
});

React.useEffect = jest.fn((c) => c());

describe('NXTable tests', () => {
  afterEach(() => jest.clearAllMocks());
  describe('Snapshot', () => {

    const DATA = [
      {
        key: '1',
        name: {
          text: 'Test name',
        },
        clients: {
          profession: 'Test profession',
        },
      },
    ];

    const COLUMNS: Column[] = [
      {
        name: 'Name',
        type: 'text',
      },
      {
        name: 'Profession',
        type: 'text',
      },
    ];

    it('should render with default props', () => {
      const component = render.create(<NXTable data={DATA} columns={COLUMNS} />);
      expect(component).toMatchSnapshot();
    });
  });
});
