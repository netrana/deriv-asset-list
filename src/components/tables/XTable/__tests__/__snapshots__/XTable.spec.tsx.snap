// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NXTable tests Snapshot should render with default props 1`] = `
<Table
  className="table"
  columns={
    Array [
      Object {
        "align": "left",
        "dataIndex": "name",
        "key": "name",
        "render": [Function],
        "title": <Header
          name="Name"
        />,
        "width": undefined,
      },
      Object {
        "align": "left",
        "dataIndex": "profession",
        "key": "profession",
        "render": [Function],
        "title": <Header
          name="Profession"
        />,
        "width": undefined,
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "clients": Object {
          "profession": "Test profession",
        },
        "key": "1",
        "name": Object {
          "text": "Test name",
        },
      },
    ]
  }
  locale={
    Object {
      "emptyText": undefined,
    }
  }
  pagination={false}
  scroll={
    Object {
      "x": true,
    }
  }
/>
`;
